src/agent/src/rpc.rs

```diff
@@ -258,8 +258,6 @@ impl AgentService {
             // Find the sandbox storage used by this container
             let mounts = sandbox.container_mounts.get(&cid);
             if let Some(mounts) = mounts {
-                remove_mounts(mounts)?;
-
                 for m in mounts.iter() {
                     if sandbox.storages.get(m).is_some() {
```

Comment:
@Burning1020 After you check that removing the mounts is a correct behavior, it would be better to change this code as follows.

```rust
if sandbox.storages.get(m).is_some() {
    sandbox.unset_and_remove_sandbox_storage(m)?;
}
```

I think `cmounts` var is no longer needed and you can also remove 268-270 lines.